packages/frontend/core/src/modules/navigation/entities/navigator.ts

```diff
@@ -0,0 +1,55 @@
+import { LiveData } from '@toeverything/infra';
+import type { Location } from 'history';
+import { Observable, switchMap } from 'rxjs';
+
+import type { Workbench } from '../../workbench';
+
+export class Navigator {
+  constructor(private readonly workbench: Workbench) {}
+
+  private readonly history = this.workbench.activeView.map(
+    view => view.history
+  );
+
+  private readonly location = LiveData.from(
+    this.history.pipe(
+      switchMap(
+        history =>
+          new Observable<{ index: number; entries: Location[] }>(subscriber => {
+            subscriber.next({ index: history.index, entries: history.entries });
+            return history.listen(() => {
+              subscriber.next({
+                index: history.index,
+                entries: history.entries,
+              });
+            });
+          })
+      )
+    ),
+    { index: 0, entries: [] }
+  );
+
+  readonly backable = this.location.map(
```

Comment:
does this work for the browser?

Reply:
The browser does not use this.
back&forward buttons are hidden on the browser